src/entities/Sto.ts

```diff
@@ -98,40 +96,82 @@ export interface Params {
  * Abstract class used as a base to manage sto functionalities
  */
 export abstract class Sto<P> extends Entity<P> {
+  /**
+   * Uniquely generated id for the STO
+   */
   public abstract uid: string;
 
+  /**
+   * Ethereum address for the STO
+   */
   public address: string;
 
   public securityTokenSymbol: string;
 
   public securityTokenId: string;
 
+  /**
+   * Type of STO setup
+   */
   public stoType: StoType;
 
   public startDate: Date;
 
   public endDate: Date;
 
+  /**
+   * Wallet where funds raised will be forwarded to
+   */
   public raisedFundsWallet: string;
 
+  /**
+   * Wallet where unsold tokens will be returned to
+   */
   public unsoldTokensWallet: string;
 
+  /**
+   * Amount of funds that have been raised so far
+   */
   public raisedAmount: BigNumber;
 
+  /**
+   * Total number of tokens that have been sold so far
+   */
   public soldTokensAmount: BigNumber;
 
+  /**
+   * Number of investors that have purchased tokens in the STO
+   */
   public investorCount: number;
 
+  /**
+   * Valid currencies that funds can be raised in
+   */
   public fundraiseCurrencies: Currency[];
 
+  /**
+   * Whether the STO is currently paused or not
+   */
   public isPaused: boolean;
 
+  /**
+   * Whether the STO cap has been reached or not
+   */
   public capReached: boolean;
 
+  /**
+   * Whether the STO has been finalized or not
+   */
   public isFinalized: boolean;
 
+  /**
+   * Whether the preissuing of tokens is allowed or not
```

Comment:
whether all tokens due to be sold are issued when the STO starts. If false, the appropriate amount of tokens are issued to the buyer whenever a sale is made